### PR TITLE
Update Inflector to new Version 1.4 || 2.0

### PR DESCRIPTION
**Pull request by bramtervoort**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no 
| Fixed tickets |  - 
| Related issues/PRs |   - 
| License | MIT
| Documentation PR | -

#### What's in this PR?

Fixed deprecation messages for Doctrine\Common\Inflector\Inflector these calls now get the new Inflecto api

#### Why?

This should make future updates a little easier.

#### To Do

- [x] update composer.json "doctrine/inflector": "^1.4 || ^2.0",

### CODE REVIEW

**Comment by alexander-schranz** [comment]
@bramtervoort Thx for the PR. Can you target the `2.1` branch for this change? And changing the `composer.json` is fine. We need here `^1.4.1 || ^2.0.1` (we can so remove the conflicts to 1.4.0 and 2.0.0 as they are not longer in the range).
This is no breaking change as we have a requirement to `handcraftedinthealps/rest-routing-bundle` which defines min version of `doctrine/inflector` to `1.4`.

**Comment by bramtervoort** [comment]
Thank you. I rebased it. if there is anything else needed i'd be haopy to do that.

File: src/Sulu/Bundle/CoreBundle/Entity/ApiEntity.php [hunk 1]
```diff
@@ -81,7 +81,9 @@ public function createSelfLink()
         // if no apiPath is not set generate it from basepath
         if (\is_null($this->getApiPath())) {
             $class = \explode('\\', \get_class($this));
-            $plural = Inflector::pluralize(\strtolower(\end($class)));
+            // this is deprecated already so no use to do fancy to get the factory
+            $inflector = InflectorFactory::create()->build();
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/CoreBundle/Entity/ApiEntity.php]: Would remove this comment does more confuse me as it helps 🙈 

```suggestion
            $inflector = InflectorFactory::create()->build();
```

File: src/Sulu/Component/Content/Metadata/Factory/StructureMetadataFactory.php [hunk 2]
```diff
@@ -52,6 +52,11 @@ class StructureMetadataFactory implements StructureMetadataFactoryInterface
      */
     private $cache = [];
 
+    /**
+     * @var \Doctrine\Inflector\Inflector
```
Comment (alexander-schranz) [inline src/Sulu/Component/Content/Metadata/Factory/StructureMetadataFactory.php]: add here `use \Doctrine\Inflector\Inflector;` for this:

```suggestion
     * @var Inflector
```

File: src/Sulu/Component/DocumentManager/Subscriber/Behavior/Path/AliasFilingSubscriber.php [hunk 3]
```diff
@@ -28,13 +28,19 @@ class AliasFilingSubscriber extends AbstractFilingSubscriber
      */
     private $metadataFactory;
 
+    /**
+     * @var \Doctrine\Inflector\Inflector
```
Comment (alexander-schranz) [inline src/Sulu/Component/DocumentManager/Subscriber/Behavior/Path/AliasFilingSubscriber.php]: same as above, add here `use \Doctrine\Inflector\Inflector;` for this:

```suggestion
     * @var Inflector
```

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
Some little change requests. Else it looks fine from my side.

**Comment by bramtervoort** [comment]
Just applied the suggestions. Github does make that very easy. Curious if this work correctly with the tests since I did not verify the php cs of this.

**Review by alexander-schranz — APPROVED** [review]

File: src/Sulu/Component/Content/Metadata/Factory/StructureMetadataFactory.php [hunk 4]
```diff
@@ -52,6 +52,11 @@ class StructureMetadataFactory implements StructureMetadataFactoryInterface
      */
     private $cache = [];
 
+    /**
+     * @var Inflector
```
Comment (alexander-schranz) [inline src/Sulu/Component/Content/Metadata/Factory/StructureMetadataFactory.php:57]: Looks for me like the `use` statement is missing here for this class

File: src/Sulu/Component/DocumentManager/Subscriber/Behavior/Path/AliasFilingSubscriber.php [hunk 5]
```diff
@@ -28,13 +28,19 @@ class AliasFilingSubscriber extends AbstractFilingSubscriber
      */
     private $metadataFactory;
 
+    /**
+     * @var Inflector
```
Comment (alexander-schranz) [inline src/Sulu/Component/DocumentManager/Subscriber/Behavior/Path/AliasFilingSubscriber.php:33]: Looks for me like the `use` statement is missing here for this class

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
@bramtervoort you can also just edit the files I just write the suggestion to maybe make it easier for you but if manual change it does work better for you use that. Currently I think the use statements are missing so the IDE would fail to find the Inflector class.

**Comment by bramtervoort** [comment]
yes it does work better in the editor. Thank you. i updated the use statements

**Review by alexander-schranz — APPROVED** [review]

**Comment by alexander-schranz** [comment]
@bramtervoort Thank your for upgrading this! 🚀

**Pull request merged**